File: Arrays/Best Time to Buy and Sell Stock II.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0;
        for(int i = 1; i < size(prices); i++){
            if(prices[i] > prices[i-1]){
                profit += prices[i] - prices[i-1];
            }
        }
        return profit;
    }
};

File: Arrays/Best Time to Buy and Sell Stock II.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut start = prices[0];
        let mut res = 0;

        for i in 1..prices.len() {
            if prices[i-1] < prices[i] {
                res += prices[i] - prices[i-1];
            }
        }
        res
    }
}